### DISCUSSION TITLE
No Build tasks in VSCode

### DISCUSSION THREAD

**Discussion by Ron5s** [Kode/Kha#1312] [General] [answered]
Last night I was able to see all the available build tasks in VSCode, but today they are gone and when I initialize a new project (F1) pressing f5 does not run the debugger, it looks like the .vscode file is not created.  Not sure if this is an extension issue or VSCode issue or both.  I am a little perplexed by this considering I did nothing to the computer overnight.

Thanks for the help,
Ron

**Comment by RobDangerous** [comment]
Happens to me as well, looks like vscode doesn't even activate the Kha extension anymore.

**Reply by Ron5s** [reply]
I see.  Thank you for letting me know. Hopefully this will get fixed soon.

**Comment by juakob** [comment]
Same here, on Wednesday we did 10 installation and all work fine. Today it stop working on my machine too. Same here F1 kha init looks like the only working command. Did a clean install, but nothing seems to work. Can't build old projects ether now.

**Reply by Ron5s** [reply]
I'm glad I'm not the only one.  Thanks for replying.

**Comment by RobDangerous** [comment, marked as answer]
Fixed in Kode/vscode-kha@5c8743ad, update already published. Yesterday's vshaxe update destroyed an old hack of mine. Serves me well is what @Gama11 secretly thinks for sure.

**Reply by Ron5s** [reply]
That works.  Thank you for the quick turn around.  Although it seems that projects generated prior to this update a have issues.  But it might be as simple as generating a new project and coping over the source code.  Ill have to see.  

![image](https://user-images.githubusercontent.com/7084944/111893861-18a49780-89c3-11eb-9ba3-ef8af99e6f57.png)

**Reply by Gama11** [reply]
Huh, interesting... [I actually insisted on measures](https://github.com/vshaxe/vshaxe/pull/479#discussion_r590432854) to avoid abuse like that for the _newly introduced_ `haxelibExecutable` API, but I thought the old `haxeExectuable` API would continue to work as-is.

But yes, I have to admit - while this wasn't intentional (I would have given you forewarning), I'm definitely happy that the Kha extension now uses the proper API for this, which should work much better for several reasons. ;)

**Comment by RobDangerous** [comment]
Please delete the .vscode folder in the project and see if that helps.

**Reply by Ron5s** [reply]
Unfortunately that does not help. :(

**Comment by RobDangerous** [comment]
Interesting. Are you sure that's related to today's changes? There also was a rework in how electron debugging works two weeks ago. And could you maybe send me a project folder that behaves like that?

PS: Getting some sleep now so the next answers have to wait a little.

**Reply by Ron5s** [reply]
No worries, sleep well.  Ill put up a small base project that exhibits the problem.  Everything worked fine before the issues I had today.  Thank you for the help.

Ron

**Reply by Ron5s** [reply]
Here is an old base project that exhibits the problem I described.  Thank you.

[Base Project.zip](https://github.com/Kode/Kha/files/6176861/Base.Project.zip)

**Comment by RobDangerous** [comment]
Works for me... but that gave me an idea: Update Kha, the changes from two weeks ago required a change to Kha.

**Reply by Ron5s** [reply]
So I updated Kha, but it had no effect.  I also uninstalled everything including the extensions, I then created  a new project and copied over the sources folder from the project I ziped and attached to this thread.  That did not work.  The problem is the build path was not set, it was empty.  I typed build and now it all works fine.  Not sure why that was not filled in though.  Thanks for all of  your help.

Ron

**Comment by RobDangerous** [comment]
Build path was not set - how and where? And typing build, what does that mean?

**Reply by Ron5s** [reply]
Im not sure how it was not set, but it was not set.  But I added "build" as the relative path here:

![image](https://user-images.githubusercontent.com/7084944/111924299-07aa6380-8a61-11eb-872e-778118cc3404.png)

**Comment by RobDangerous** [comment]
Ah, guess you accidentally changed it then when you tried to clean things up.

**Reply by Ron5s** [reply]
Yea that would probably be the case.  Thanks for the help.
